src/tree/AttachedAccountsTreeItem.ts

```diff
@@ -100,21 +102,53 @@ export class AttachedAccountsTreeItem implements IAzureParentTreeItem {
 
             if (connectionString) {
                 let treeItem: IAzureTreeItem = await this.createTreeItem(connectionString, defaultExperience);
-                if (this._attachedAccounts.find(s => s.id === treeItem.id)) {
-                    vscode.window.showWarningMessage(`Database Account '${treeItem.id}' is already attached.`)
-                } else {
-                    this._attachedAccounts.push(treeItem);
-                    if (this._keytar) {
-                        await this._keytar.setPassword(this._serviceName, treeItem.id, connectionString);
-                        await this.persistIds();
-                    }
-                }
+                await this.attachAccount(treeItem, connectionString);
             }
         } else {
             throw new UserCancelledError();
         }
     }
 
+    public async attachEmulator(): Promise<void> {
+        let connectionString: string;
+        const defaultExperience = <Experience>await vscode.window.showQuickPick(['MongoDB', 'DocumentDB', 'Graph'], { placeHolder: "Select a Database Account API...", ignoreFocusOut: true });
+        if (defaultExperience) {
+            let port: number;
+            if (defaultExperience === Experience.MongoDB) {
+                port = vscode.workspace.getConfiguration().get<number>("cosmoDB.emulator.mongoPort") || 10255;
+            }
+            else {
+                port = vscode.workspace.getConfiguration().get<number>("cosmosDB.emulator.port") || 8081;
+            }
+            if (port) {
+                if (defaultExperience === Experience.MongoDB) {
+                    connectionString = `mongodb://localhost:C2y6yDjf5/R+ob0N8A7Cgv30VRDJIWEHLM+4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw/Jw==@localhost:${port}/admin?ssl=true`;
```

Comment:
This is the connection string for the 'admin' database. I would suggest removing 'admin' so that it's just the connection string for the mongo account. I believe that would help simplify your connection string logic in MongoDatabaseTreeItem